### PR TITLE
Deferred restart test fixes

### PR DESCRIPTION
**Pull request by gnuoy**

### CODE REVIEW

File: zaza/openstack/charm_tests/test_utils.py [hunk 1]
```diff
@@ -728,16 +728,10 @@ def setUpClass(cls):
     """
 
     @classmethod
-    def setUpClass(cls, restart_config_file, test_service, restart_package,
-                   restart_package_service, application_name):
+    def setUpClass(cls, restart_package, restart_package_service,
+                   application_name, deferred_hook='config-changed'):
```
Comment (AurelienLourot) [inline zaza/openstack/charm_tests/test_utils.py]: `deferred_hook` is in fact a string that will be searched in the workload status msg and I think this deserves a docstring

**Review by AurelienLourot — CHANGES_REQUESTED** [review]
LGTM in general, just one nitpick, thanks!

File: zaza/openstack/charm_tests/test_utils.py (hunk 1 above)
Comment (AurelienLourot) [reply zaza/openstack/charm_tests/test_utils.py]: Oh it's gone meanwhile, sorry. I have been reviewing commit by commit. Let me take a final look

File: zaza/openstack/charm_tests/test_utils.py [hunk 2]
```diff
@@ -847,107 +843,202 @@ def check_show_deferred_events_action_hook(self, hook):
...
-            {config_key: new_value})
+            restart_config_file,
+            expected_contents)
+        logging.info("Waiting for units to be idle")
+        model.block_until_all_units_idle()
+
+    def trigger_deferred_hook_via_charm(self, deferred_hook):
+        """Set charm config option which requires a service start.
+
+        Set the charm debug option and wait for that change to be renderred in
```
Comment (AurelienLourot) [inline zaza/openstack/charm_tests/test_utils.py:930]: s/renderred/rendered/

File: zaza/openstack/charm_tests/test_utils.py [hunk 3]
```diff
@@ -847,107 +843,202 @@ def check_show_deferred_events_action_hook(self, hook):
...
+        applications config file.
+
+        :param deferred_hook: Hook or method name which should be showing as
+                              defeerred.
```
Comment (AurelienLourot) [inline zaza/openstack/charm_tests/test_utils.py:934]: also defeerred typo here

**Review by AurelienLourot — CHANGES_REQUESTED** [review]
Only typos, thanks!

**Review by AurelienLourot — APPROVED** [review]
LGTM, thanks! Not merging it just yet in order to give @ajkavanagh a chance to review it as well.

**Review by ajkavanagh — APPROVED** [review]
LGTM

**Pull request merged**